elide-core/src/main/java/com/yahoo/elide/core/FilterScope.java

```diff
@@ -27,6 +29,23 @@ public FilterScope(RequestScope requestScope, Class<?> resourceClass) {
         this.permissions = requestScope.getDictionary().getPermissionsForClass(resourceClass, ReadPermission.class);
     }
 
+    /*
+     *This constructor with FilterExpression allows merging the filterExpression parsed from URL and the one from
+     *FilterExpressionCheck.
+     */
+    public FilterScope(RequestScope requestScope, Class<?> resourceClass, FilterExpression permissionExpression) {
```

Comment:
You don't need a new constructor here.  The filterExpression can be derived from the PermissionExecutor which can be retrieved from the RequestScope.